### PR TITLE
Integrate application management endpoint - Closes #4489

### LINKED ISSUE #4489: Integrate application management endpoint
### Description
application management is running by mock data and it's time to adopt with v6 endpoint

### Acceptance Criteria
- fetch application default by useBlockchainApplicationMeta and merge with application from redux
- add/update status of the application list from server
- adopt application response changes to the app
- adopt unit test

### PR DESCRIPTION
**Pull request by Balanced02**
### What was the problem?

This PR resolves #4489 

### How was it solved?

- [x] Integrate Applications Management with api
- [x] Integrate Applications Statistics with api
- [x] Integrate Application Details with api
- [x] Ensure adding and switching applications work

### How was it tested?

- [x] Browser
- [x] Updated/Added Unit tests

### CODE REVIEW

File: src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js [hunk 1]
```diff
@@ -9,36 +9,47 @@ import grid from 'flexboxgrid/dist/flexboxgrid.css';
...
   const chainId = parseSearchParams(location.search).chainId;
   const mode = parseSearchParams(location.search).mode;
   const { checkPinByChainId, togglePin } = usePinBlockchainApplication();
-  const {
-    name, state, lastCertificateHeight, lastUpdated,
-  } = application.data;
   const { setApplication } = useApplicationManagement();
 
+  // TODO: Replace chainID here with chainID from location.search
+  const application = useBlockchainApplicationMeta({ config: { params: { chainID: '00000001' } } })
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js:33]: Kindly, please fix the chain id from location

File: src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js [hunk 2]
```diff
@@ -85,8 +96,8 @@ const BlockchainApplicationDetails = ({ history, location, application }) => {
             <a
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js:96]: https://docs.google.com/document/d/1fId3uEv1xLfXxBI3sipIGPWl4YiN9caheuQIRZ68VVk/edit#
```suggestion
          blockchain.explorers.map(({url}) =>  <a
              className={`${styles.appLink}`}
              target="_blank"
              href={url}
            >
              <Icon name="chainLinkIcon" className={styles.hwWalletIcon} />
              {url}
            </a>)
```

File: src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// import { useCustomInfiniteQuery } from 'utilities/api/hooks/useCustomInfiniteQuery';
+import { GET_APPLICATION_QUERY } from 'src/const/queries';
+import { useCustomInfiniteQuery } from 'src/modules/common/hooks';
+import { LIMIT, API_URL, API_BASE_URL } from 'src/utils/api/constants';
+
+/**
+ * Fetch list of blockchain applications on-chain data.
+ * Executes the API call once the hook is mounted.
+ * @param {Object} config - Custom configurations for the query.
+ * @param {Object} options - Custom options for the query.
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.js]: please update docs with params

File: src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const mockStore = configureMockStore();
+
+const ReduxProvider = ({ children, reduxStore }) => (
+  <Provider store={reduxStore}>{children}</Provider>
+);
+
+describe('useApplicationsQuery hook', () => {
+  it('should fetch data correctly', async () => {
+    const store = mockStore(mockState);
+    const wrapper = ({ children }) => (
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js]: please use queryWrapper 
```
import { queryWrapper as wrapper } from 'src/utils/test/queryWrapper';
```

File: src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    current: mockBlockchainApp[0],
+    pins: [],
+  },
+};
+
+const queryClient = new QueryClient();
+const mockStore = configureMockStore();
+
+const ReduxProvider = ({ children, reduxStore }) => (
+  <Provider store={reduxStore}>{children}</Provider>
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js]: useQuery nothing to do with redux

File: src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * @returns - The query state of the API call. Includes the data
+ * (applications), loading state, error state, and more.
+ */
+export default function useApplicationsQuery(chainID, { config: customConfig = {}, options = {} } = {}) {
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.js]: kindly please change the `useApplicationsQuery` to `useApplications` 
to match with the rest of hooks

File: src/modules/blockchainApplication/explore/hooks/queries/useBlockchainApplicationExplore.js [hunk 7]
```diff
@@ -1,40 +1,24 @@
...
-// eslint-disable-next-line import/prefer-default-export
-export const useBlockchainApplicationExplore = ({ config: customConfig = {}, options } = {}) => {
-  const config = {
-    url: `/api/${API_VERSION}/blockchain/apps`,
-    method: 'get',
-    event: 'get.blockchain.apps',
-    ...customConfig,
-    params: { limit, ...(customConfig?.params || {}) },
-  };
+export default function useBlockchainApplicationExplore() {
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/hooks/queries/useBlockchainApplicationExplore.js:9]: here its need to call the meta endpoint to add meta data inside the application

File: src/modules/blockchainApplication/explore/hooks/queries/useBlockchainApplicationExplore.js [hunk 8]
```diff
@@ -1,40 +1,24 @@
...
+  const applications = useMemo(() => {
+    const data = applicationsQuery.data?.data.map((app) => ({
+      ...app,
+      isPinned: checkPinByChainId(app.chainID),
+    }));
+
+    return { ...applicationsQuery, data, meta: applicationsQuery.data?.meta };
+  }, [applicationsQuery, pins, checkPinByChainId]);
+
+  return applications
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/hooks/queries/useBlockchainApplicationExplore.js:23]: the pagination is missing

File: src/modules/blockchainApplication/explore/hooks/queries/useBlockchainApplicationStatistics.js [hunk 9]
```diff
@@ -1,35 +1,19 @@
...
-    method: 'get',
-    event: 'get.blockchain.apps.statistics',
-    ...customConfig,
-  };
-  return useCustomQuery(
-    {
-      keys: [BLOCKCHAIN_APPS_STATICS],
-      config,
-      options,
+export default function useBlockchainApplicationStatistics() {
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/hooks/queries/useBlockchainApplicationStatistics.js:5]: we are not using export default any more

File: src/modules/blockchainApplication/manage/components/AddApplicationList/AddApplicationList.js [hunk 10]
```diff
@@ -5,35 +5,34 @@ import Box from '@theme/box';
...
     : false;
 
   const handleLoadMore = () => {
     const params = {
       ...filters,
-      offset: liskApplications.meta.count + liskApplications.meta.offset,
+      offset: applications.meta.count + applications.meta.offset,
     };
-    liskApplications.loadData(params);
+    fetchNextPage(params)
```
Comment (soroushm) [inline src/modules/blockchainApplication/manage/components/AddApplicationList/AddApplicationList.js:35]: fetchNextPage dosn't need params its already handled inside the infiniteQuery

File: src/modules/blockchainApplication/manage/components/AddApplicationList/AddApplicationList.test.js [hunk 11]
```diff
@@ -1,69 +1,44 @@
...
-  address: 'lsk24cd35u4jdq8ssd03nfmte5dsxwrnazyqqqg5eu',
-  lastCertificateHeight: 400,
-  lastUpdated: 123456789,
-  depositedLsk: 820000000,
-};
+const mockFetchNextPage = jest.fn()
+
+jest.mock('src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery');
+
+useApplicationsQuery.mockReturnValue({
```
Comment (soroushm) [inline src/modules/blockchainApplication/manage/components/AddApplicationList/AddApplicationList.test.js]: please avoid mocking queries lets return data from msw

File: src/utils/api/constants.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+export const API_BASE_URL = 'http://165.22.29.229:9901';
```
Comment (soroushm) [inline src/utils/api/constants.js:1]: we have this data in `src/const/config` pleaase avoid dublication

**Review by soroushm — CHANGES_REQUESTED** [review]

File: src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js (hunk 5 above)
Comment (Balanced02) [reply src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js]: ...the `useInfiniteQuery` needs to get `currentApplication` from redux; other option is to mock the implementation of `useCustomInfiniteQuery`

File: src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js (hunk 4 above)
Comment (Balanced02) [reply src/modules/blockchainApplication/explore/hooks/queries/useApplicationsQuery.test.js]: ...same reason above; needs redux

**Comment by Balanced02** [comment]
Did a re-work in a different branch

**Pull request closed without merging**